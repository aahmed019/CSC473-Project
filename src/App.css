.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.middle{
  background: linear-gradient(90deg, #01000e 0%, #4157a0 50%, #01000e 100%);
  
}

.Welcome-Section{
  height: 10vh;
}

.Body-Section{
  background: linear-gradient(90deg, #01000e 0%, #4157a0 50%, #01000e 100%);
}

.popular-game{
  text-align: center;
  font-size: 40px;
  color: white;
  text-decoration: underline;
}

.bodyCard{
  list-style-type: none;
  font-size: 20px;
  text-align-last: center;
  overflow: hidden;
  /* align: left; */
  /* padding: 5em; */
}
.card{
  align-content: center;
  width: 80vw;
  margin-top: 2vh;
  left: 5%;
}


.card-title{
  font-size:30px;
}
.card-img-top{
  width: 70vw;
  height: 10px;

}
  
.card-img-top{
    width: 50%;
    height: 5%;
  }

.footer-all {
  bottom:0;
  /*background-color: #292c2f;*/
  background-color: black;
  box-sizing: border-box;
  width: 100%;
  font: normal 100px sans-serif;
  padding: 45px 50px;
  /* margin-top: 80px; */
  
}

.footer-social {
  margin: 0 auto 20px;
}

.footer-social a {
  display: inline-block;
  width: 10%;
  /*height: 10%;*/
  border-radius: 2px;
  font-size: 50%;
  color: white;
  line-height: 38px;
  margin-left: 3px;
}

.footer-info {
  font-size: 35%;
  font-weight: bold;
  color: #ffffff;
}
.footer-info a{
  color:white;
  padding:10px;
  text-decoration: none;
}


.footer-all .footer-left, .footer-distributed .footer-right {
  text-align: center;
}

/* 
Everything below this point is strictly for the game page
The seperation is for if we want to move it to a different file

*/


.textbox{
width:50%;
font-size:30px;
text-align: center;
}

.g-image{

  align-content: center;
  margin-top:10%;
  margin-left: 15%;
  width: 70vw;
  height: 25vh;
}


.btn-large{
  font-size: 3rem;
  
}

.exit{
  margin-left: 5%;
}
.cr{
  padding:10%;
  font-size: 300%;
}
.hint{
  font-size:200%;
}
.questN{
  font-size: 200%;
  font-style:italic ;
}
.quest{
  font-size: 250%;
}

.navbarGame{
    background-color: white;
    top: 0;
    left: 0;
    width: 100%;
    height: 7vh;
    border-style: solid;
    border-color: black;

}
.toggleGame-button-line {
  width: 2vh;
  height: 0.25vh;
  background: black;
}
.spaceGame-btw-logo-items {
  flex: .95;
}
.gameTitle{
  text-decoration: underline;
}
img {
  width: 80%;
  height: auto;
}