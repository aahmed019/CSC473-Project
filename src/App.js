import React, { Component } from 'react';
import games from './games.json';
import './App.css';
import Navbar from './components/Navbar/Navbar';
import SideDrawerMenu from './components/SideDrawerMenu/SideDrawerMenu';
import Backdrop from './components/Backdrop/Backdrop';
// import sherapd from './images/sherapd0.jpg'

class App extends Component {
  state = {
    sideDrawerMenuOpen: false
  };

  // toggle drawer button handler
  drawerToggleClickHandler = () => {
    this.setState((prevState) => {
      return {sideDrawerMenuOpen: !prevState.sideDrawerMenuOpen};
    })
  };

  // clicking backdrop closes side drawer
  backdropClickHandler = () => {
    this.setState({sideDrawerMenuOpen: false});
  }
  
  // this state later will be written into a file in a backend so creator can generate new games without coding

  render = () => {
    // console.log(games) 
    let listItems = games
      .filter(game => game.popular == true)
      .map(item =>
        <li className="bodyTitle" key={item.id} >
          <strong>{item.Title} </strong>
          <strong>({item.Difficulty}/5)</strong>
          <br />
          <img
            key={item.id}
            className={("img" + item.id)}
            src={require('./images/' + item.thumbnail)}
            onMouseEnter={e => e.currentTarget.src = require('./images/' + item.id + "/1.jpg")}
            onMouseOut={e => e.currentTarget.src = require('./images/' + item.thumbnail)}
          />
        </li >
      );
    
    // handle side drawer 
    let backdrop;
    if (this.state.sideDrawerMenuOpen) {
      backdrop = <Backdrop click = { this.backdropClickHandler }/>;
    }
    
    return (
      <div className="App">
         <Navbar drawerClickHandler = { this.drawerToggleClickHandler }/>;
         <SideDrawerMenu show = { this.state.sideDrawerMenuOpen }/>;
         { backdrop };                           

                                         
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
          integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
          crossorigin="anonymous" />
        <link rel="stylesheet" href="styles.css"></link>
        <header>
          <h1 className="greeting">
            <br />
          </h1>
        </header>
        <body className="body">
          <p><strong>Popular Games</strong></p>
          <ol >{listItems}</ol>
        </body>


        <footer className="footer-all">

			    <div className="footer-social">

				<a href="#"><i className="fab fa-facebook"></i></a>
				<a href="#"><i className="fab fa-twitter"></i></a>
				<a href="#"><i className="fab fa-linkedin"></i></a>
				<a href="#"><i className="fab fa-github"></i></a>

			  </div>

			<div className="footer-info">

				<p>
					<a href="#">Home</a>
					·
					<a href="#">Games</a>
					·
					<a href="#">About</a>
					·
					<a href="#">Faq</a>
					·
					<a href="#">Contact</a>
				</p>

				<p><strong>Escape Team © 2019</strong></p>
			</div>

		</footer>
      </div>
    );
  }
}


export default App;