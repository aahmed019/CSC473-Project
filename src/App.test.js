import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

// https://medium.com/codeclan/testing-react-with-jest-and-enzyme-20505fec4675
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('backdropClickHandler Testing',()=>{
  let component=renderer.create(<App />).getInstance();

  component.backdropClickHandler();

  expect(component.state.sideDrawerMenuOpen).toBeFalsy;

})
test('drawerToggleClickHandler Testing',()=>{
  let component=renderer.create(<App />).getInstance();

  component.drawerToggleClickHandler();

  expect(component.state.sideDrawerMenuOpen).toHaveReturned;

})
const puppeteer=require('puppeteer');

test('testing hero of angle game',async() => {
    const browser = await puppeteer.launch({
        // headless: false,
        // slowMo: 180,
        // args:['window-size = 1920,1080']
    });
    const page = await browser.newPage();
    await page.goto('https://master.dlhem6nvy7qu4.amplifyapp.com');
    await page.click('button.login');
    await page.click('input.Input__input___3e_bf');
    await page.type('input.Input__input___3e_bf','admin2');
    await page.click('[type=password]');
    await page.type('[type=password]','admin12345')
    await page.click('button.Button__button___vS7Mv');
    await page.click('button.toggle-button');
    await page.click('div.toggle-button-line');
    await page.click('[href="/Game"]');
    
    await page.click('#bttn1')
    await page.click('#start-btn')
    let game1=await page.$eval('h1',(e)=>e.textContent);  
    expect(game1).toBe('Hero of the Angle Challenge');

    // testing the hero of the angle game
    await page.click('button#num1');
    await page.click('button#num8');
    await page.click('button#num6');
    await page.click('button#num9');
    await page.click('button#pound');
    let temp = await page.$eval('div#textAnswer', (div) => div.id);
    expect(temp).toBe('textAnswer');
    
    await page.click('input#answerBox');
    await page.type('input#answerBox', 'time');
    await page.click('button#submitBttn');
    temp = await page.$eval('div#textAnswer', (div) => div.id);
    expect(temp).toBe('textAnswer');

    await page.click('input#answerBox');
    await page.type('input#answerBox', 'cane');
    await page.click('button#submitBttn');
    temp = await page.$eval('div#textAnswer', (div) => div.id);
    expect(temp).toBe('textAnswer');

    await page.click('input#answerBox');
    await page.type('input#answerBox', 'horace webster');
    await page.click('button#submitBttn');
    temp = await page.$eval('div#numPad', (div) => div.id);
    expect(temp).toBe('numPad');

    await page.click('button#num7');
    await page.click('button#pound');
    
    await page.click('button#num1');
    await page.click('button#num2');
    await page.click('button#pound');

    await browser.close();
},20000);

