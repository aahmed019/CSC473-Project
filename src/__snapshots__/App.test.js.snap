// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="App"
>
  <link
    crossOrigin="anonymous"
    href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
    integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
    rel="stylesheet"
  />
  <link
    href="styles.css"
    rel="stylesheet"
  />
  <meta
    content="600px"
    name="viewport"
  />
  <navbar
    drawerClickHandler={[Function]}
  />
  <sideDrawerMenu
    show={false}
  />
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        path="/Home"
      />
      <Route
        component={[Function]}
        path="/Game"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/contact"
      />
    </Switch>
  </BrowserRouter>
  <footer />
</div>
`;

exports[`should render banner text correctly with given strings 1`] = `
<div
  className="App"
>
  <link
    crossOrigin="anonymous"
    href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
    integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
    rel="stylesheet"
  />
  <link
    href="styles.css"
    rel="stylesheet"
  />
  <meta
    content="600px"
    name="viewport"
  />
  <navbar
    drawerClickHandler={[Function]}
  />
  <sideDrawerMenu
    show={false}
  />
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        path="/Home"
      />
      <Route
        component={[Function]}
        path="/Game"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/contact"
      />
    </Switch>
  </BrowserRouter>
  <footer />
</div>
`;
